arlas-core/src/main/java/io/arlas/server/dao/ElasticCollectionReferenceDaoImpl.java

```diff
@@ -153,59 +153,24 @@ public void deleteCollectionReference(String ref) throws NotFoundException, Inte
-            if (collectionReference.params.timestampPath != null)
-                fields.add(collectionReference.params.timestampPath);
-            if(collectionReference.params.excludeFields!=null && collectionReference.params.excludeFields!=""){
-                List<String> excludeField = Arrays.asList(collectionReference.params.excludeFields.split(","));
-                CheckParams.checkExcludeField(excludeField, fields);
-            }
-
-            Iterator<String> indeces = response.getMappings().keysIt();
-            while (indeces.hasNext()) {
-                String index = indeces.next();
-                //check type
-                try {
-                    if (!response.getMappings().get(index).containsKey(collectionReference.params.typeName)) {
-                        throw new NotFoundException("Type " + collectionReference.params.typeName + " does not exist in " + collectionReference.params.indexName + ".");
-                    }
-                    Object properties = response.getMappings().get(index).get(collectionReference.params.typeName).sourceAsMap().get("properties");
-                    if (properties == null) {
-                        throw new NotFoundException("Unable to find properties from " + collectionReference.params.typeName + " in " + index + ".");
-                    }
-                } catch (Exception e) {
-                    throw new NotFoundException("Unable to get " + collectionReference.params.typeName + " in " + index + ".");
-                }
-
-                //check fields
-                if (!fields.isEmpty()) {
-                    ElasticTool.checkIndexMappingFields(client, index, collectionReference.params.typeName, fields.toArray(new String[fields.size()]));
-                    setTimestampFormatOfCollectionReference(index, collectionReference.params);
-                }
-
-
-            }
-        } catch (ArlasException e) {
-            throw e;
-        } catch (IndexNotFoundException e) {
-            throw new NotFoundException("Index " + collectionReference.params.indexName + " does not exist.");
-        } catch (Exception e) {
-            throw new NotFoundException("Unable to access " + collectionReference.params.typeName + " in " + collectionReference.params.indexName + ".");
+        //get fields
+        List<String> fields = new ArrayList<>();
+        if (collectionReference.params.idPath != null)
+            fields.add(collectionReference.params.idPath);
+        if (collectionReference.params.geometryPath != null)
+            fields.add(collectionReference.params.geometryPath);
+        if (collectionReference.params.centroidPath != null)
+            fields.add(collectionReference.params.centroidPath);
+        if (collectionReference.params.timestampPath != null)
+            fields.add(collectionReference.params.timestampPath);
+        if(collectionReference.params.excludeFields!=null && collectionReference.params.excludeFields!=""){
```

Comment:
If you have Apache, there is a StringUtils.isEmpty or .isBlank to check a string for not null nor empty